### ISSUE TITLE
DeclarationStatement is too general

### ISSUE THREAD

**Issue by hackedy** [cornell-netlab/petr4#5]
We allow any declaration to be a statement, but we could restrict to constant declarations, variable declarations, and instantiations.

[This is in section 10.3 of the v1.1.0 spec.](https://p4.org/p4-spec/docs/P4-16-v1.1.0-spec.html#sec-block-stmt)

**Comment by jnfoster** [comment]
Yeah. @ericthewry and I noticed the same thing earlier this week...

**Comment by hackedy** [comment]
Taking another look at this, it seems like we are putting declarationstatements into the statement type itself instead of the block type, which is where they belong.
